### PR TITLE
feat(multi-action-button, split-button): pass props down to main button

### LINKED ISSUE #4972: Not possible to set ID on splitButton
### Current behaviour

It's not possible to set an ID for a splitButton.

(The ID prop isn't passed down and it just uses a GUID instead).

### Expected behaviour

Would expect and ID prop to be supported for at least the main button but ideally the dropdown toggle button and any children, too

### CodeSandbox or Storybook URL

https://codesandbox.io/s/no-id-on-splitbutton-68c09t

### JIRA Ticket (Sage Only)

_No response_

### Suggested Solution

Add support for an ID prop.

### Carbon Version

104.55.0

### What browsers are you seeing the problem on?

Chrome, Microsoft Edge

### What Operating System are you seeing the problem on?

Windows

### Anything else we should know?

_No response_

### Confidentiality

- [X] I confirm there is no confidential or commercially sensitive information included.

### PR DESCRIPTION
**Pull request by robinzigmond**
### Proposed behaviour

Allow the user to provide any HTML attributes such as "id" to the MultiActionButton and SplitButton components, and have these added to the main button element.

Fixes #4972

### Current behaviour

No such attributes are passed down. In particular, any IDs passed are ignored, and for SplitButton a guid is provided as the ID for the main button.

### Checklist

- [X] Commits follow our style guide
- [X] Related issues linked in commit messages if required
- [X] Screenshots are included in the PR if useful
- [X] All themes are supported if required
- [X] Unit tests added or updated if required
- [X] Cypress automation tests added or updated if required
- [X] Storybook added or updated if required
- [X] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [X] Typescript `d.ts` file added or updated if required

#### QA

- [ ] Tested in CodeSandbox/storybook
- [ ] Add new Cypress test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions

The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.

https://codesandbox.io/s/no-id-on-splitbutton-68c09t

Also test for MultiActionButton

### CODE REVIEW

File: src/components/multi-action-button/multi-action-button.component.js [hunk 1]
```diff
@@ -166,6 +167,7 @@ const MultiActionButton = ({
       buttonType: buttonType || as,
       size,
       subtext,
+      ...rest,
```
Comment (mkrds) [inline src/components/multi-action-button/multi-action-button.component.js:170]: Filter out all styled-system props from `rest` before spreading it here

File: src/components/split-button/split-button.component.js [hunk 2]
```diff
@@ -143,6 +145,7 @@ const SplitButton = ({
       onClick,
       size,
       subtext,
+      ...rest,
```
Comment (mkrds) [inline src/components/split-button/split-button.component.js:148]: Filter out all styled-system props from `rest` before spreading it here

**Comment by mkrds** [comment]
*.d.ts files need to be modified as well in this PR

**Review by DipperTheDan — APPROVED** [review]

**Review by edleeks87 — APPROVED** [review]

**Review by ZhuoyuJin — APPROVED** [review]

**Pull request merged**